go.mod

```diff
@@ -257,6 +256,7 @@ require (
 require (
 	github.com/KimMachineGun/automemlimit v0.6.0
 	github.com/Shopify/sarama v1.38.1
+	github.com/grafana/alloy-remote-config v0.0.3
 	github.com/grafana/kafka_exporter v0.0.0-20240409084445-5e3488ad9f9a
 )
```

Comment:
I'm not a fan of how Go tooling keeps adding so many require blocks. Can you take a second to merge the require blocks again? We should only have two: one for direct dependencies and one for indirect dependencies.

Reply:
I cleaned up so indirect deps are grouped as well.